### PR TITLE
[16.0] [MIG] email_template_config

### PR DESCRIPTION
**Pull request by victor-champonnois**
https://gestion.coopiteasy.be/web#id=10959&action=475&active_id=492&model=project.task&view_type=form&menu_id=

### CODE REVIEW

File: email_template_config/models/mail_template.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    force_email_send = fields.Boolean(string="Force mail send?")
+
+    def send_mail(
+        self,
+        res_id,
+        force_send=False,
+        raise_exception=False,
+        email_values=None,
+        notif_layout=False,
```
Comment (huguesdk) [inline email_template_config/models/mail_template.py]: [in 16.0](https://github.com/OCA/OCB/blob/16.0/addons/mail/models/mail_template.py#L314), this argument has been renamed to `email_layout_xmlid`.

**Review by huguesdk — CHANGES_REQUESTED** [review]

**Review by carmenbianca — COMMENTED** [review]
LGTM, same comment as hugues

File: email_template_config/models/mail_template.py (hunk 1 above)
Comment (victor-champonnois) [reply email_template_config/models/mail_template.py]: fixed

**Review by carmenbianca — APPROVED** [review]

**Review by robinkeunen — CHANGES_REQUESTED** [review]
I actually suspect there is something wrong with this module, I read in our migration analysis tool : 

> à garder
> Module indispensable pour les implémentations cooperator (notamment) car en standard tous les mail template sont en ""force send"". Du coup, si le serveur mail n'est pas OP, impossible de lettrer les demandes de libération."

I see many wrong things
- why does the cooperator modules depend on this module fo function properly while not declaring the dependency ?
- `force_send`'s default value is False (that is, put in the email queue). Therefore it should not block the mail unless we `send_mail(id, force_send=True)
- it should be named `mail_template_force_send` or the like

The need for this module comes from (version 12)

```py 

# cooperator/models/account_invoice.py
class AccountInvoice(models.Model):
    def set_cooperator_effective(self, effective_date):
        ...
        self.create_user(self.partner_id)

    def create_user(self, partner):
        ...
        user.sudo().with_context({"create_user": True}).action_reset_password()

# auth_signup/models/res_users.py
class ResUsers(models.Model):
    def action_reset_password(self):
        ...
        template.with_context(lang=user.lang).send_mail(user.id, force_send=True, raise_exception=True)
```

The send_mail(force_send=True) waits for the mail to be sent and blocks the process if it fails. There should be a better wait to fix this.
By using oca/queue for example.

**Comment by victor-champonnois** [comment]
- why does the cooperator modules depend on this module fo function properly while not declaring the dependency ?

**It does not strictly depend on it. It might be required for implementation for which we don't want or can't send email to cooperators**

- force_send's default value is False (that is, put in the email queue). Therefore it should not block the mail unless we `send_mail(id, force_send=True)

**and we indeed do in cooperator/models/account_invoice, line 197 : email_template.sudo().send_mail(self.id, True). I don't know why the force send is set in this line, though.**


- it should be named mail_template_force_send or the like

**Yes good idea**

**Comment by huguesdk** [comment]
note: while in version 12 and 14, there are indeed 2 mail messages sent with `force_send=True`, this argument has been removed from all `send_mail()` calls in version 16.

**Comment by robinkeunen** [comment]
@huguesdk indeed but a force_send=True is still implicitely called by `action_reset_password` : 

https://github.com/OCA/OCB/blob/b9fd7c5ac938fb0fc265abc7a4d202c3fd090b7b/addons/auth_signup/models/res_users.py#L224

**Comment by robinkeunen** [comment]
Discuté sur la tâche : ne pas porter 💩

**Pull request closed without merging**